app/components/Sidebar/Sidebar.js

```diff
@@ -26,7 +26,7 @@ function Sidebar({ location, children }: Props) {
     if (location !== previousLocation) {
       ui.hideMobileSidebar();
     }
-  }, [ui, location]);
+  }, [ui, location, previousLocation]);
```

Comment:
I had a linter warning here, it's generally unrelated to the rest of the changes